### PR TITLE
Introduces CGWindowList API for window capturing and improves `CGImage` and `CGContext`

### PR DESCRIPTION
**Pull request by application-developer-DA**
Initially I just wanted to rewrite https://github.com/servo/core-graphics-rs/pull/52 as it seems that the author is not interested in changing his pull request. So I just implemented `CGImage` bindings, but today another pull request has been merged (https://github.com/servo/core-graphics-rs/pull/79) which does exactly the same.

This pull request still improves some things in regards to `CGImage`, the most important ones are:
* Now `CGImage` uses `CGImageRelease` instead of `CFRelease` (it's a much safer to use `CGImageRelease` here, as it does not crash if you pass a `NULL` image reference to it).
* Now you can retrieve bytes from `CGImage` (uses data provider and copies the bytes from it).

Also several improvements in `CGContext`:
* Added a function `create_image()`, which allows you to create an image out of existing `CGContext`.
* Now it is possible to specify a data pointer passed as a first parameter to the `CGBitmapContextCreate()` upon `CGContext` creation -- very important thing when you deal with images as it allows you to specify your own buffer instead of allocating and copying an additional one from the context (tested it on my machine, it gave me 18.46% of performance boost).

And of course this pull requests adds `CGWindowListCreateImage()` and related things, so now window capturing is possible.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/core-graphics-rs/80)

### CODE REVIEW

File: src/geometry.rs [hunk 1]
```diff
@@ -70,15 +72,32 @@ impl CGRect {
             ffi::CGRectInset(*self, size.width, size.height)
         }
     }
+
+    #[inline]
+    pub fn from_dict_representation(dict: &CFDictionary) -> Option<CGRect> {
+        let mut rect = CGRect::new(&CGPoint::new(0., 0.), &CGSize::new(0., 0.));
+        let result = unsafe {
+            ffi::CGRectMakeWithDictionaryRepresentation(dict.as_concrete_TypeRef(),
+                                                        &mut rect as *mut CGRect)
```
Comment (jdm) [inline src/geometry.rs]: `as *mut CGRect` shouldn't be necessary.

**Comment by application-developer-DA** [comment]
Ok, done.

I rebased and squashed the commits (if I got you right, you wanted me to rebase the commits and then to do a force push) and also I took your remark and removed an unnecessary cast in `geometry.rs`.

**Pull request merged**